inst/stan/estimate_infections.stan

```diff
@@ -30,29 +30,31 @@ transformed data{
   real r_logmean = log(r_mean^2 / sqrt(r_sd^2 + r_mean^2));
   real r_logsd = sqrt(log(1 + (r_sd^2 / r_mean^2)));
 
-  int delay_max_fixed = (n_fixed_delays == 0 ? 0 :
-    sum(delay_max[fixed_delays]) - num_elements(fixed_delays) + 1);
-  int delay_max_total = (delays == 0 ? 0 :
-    sum(delay_max) - num_elements(delay_max) + 1);
-  vector[gt_fixed[1] ? gt_max[1] : 0] gt_fixed_pmf;
-  vector[truncation && trunc_fixed[1] ? trunc_max[1] : 0] trunc_fixed_pmf;
-  vector[delay_max_fixed] delays_fixed_pmf;
+  int gt_max_fixed =
+    num_elements(gt_np_pmf) - num_elements(gt_np_pmf_groups) + 1;
+  int gt_max_total = gt_max_fixed + sum(gt_max) - num_elements(gt_max);
+  int delay_max_fixed =
+    num_elements(delay_np_pmf) - num_elements(delay_np_pmf_groups) + 1;
+  int delay_max_total = delay_max_fixed + sum(delay_max) - num_elements(delay_max);
+  int trunc_max_fixed =
+    num_elements(trunc_np_pmf) - num_elements(trunc_np_pmf_groups) + 1;
+  int trunc_max_total = trunc_max_fixed + sum(trunc_max) - num_elements(trunc_max);
 
-  if (gt_fixed[1]) {
-    gt_fixed_pmf = discretised_pmf(gt_mean_mean[1], gt_sd_mean[1], gt_max[1], gt_dist[1], 1);
-  }
-  if (truncation && trunc_fixed[1]) {
-    trunc_fixed_pmf = discretised_pmf(
-      trunc_mean_mean[1], trunc_sd_mean[1], trunc_max[1], trunc_dist[1], 0
-    );
-  }
-  if (n_fixed_delays) {
-    delays_fixed_pmf = combine_pmfs(
-      to_vector([ 1 ]), delay_mean_mean[fixed_delays],
-      delay_sd_mean[fixed_delays], delay_max[fixed_delays], 
-      delay_dist[fixed_delays], delay_max_fixed, 0, 0
-    );
-  }
+  vector[gt_max_fixed] gt_fixed_pmf;
```

Comment:
this is quite a simple step - we could be doing this in the `c` method of dist vs in the stan model..

Reply:
Done now in a new `+` method for `dist_spec` (extending the now internal `c` method that won't convolute - needed to put different delays together that shouldn't be combined, e.g. generation time and reporting delays).